verilog/CST/type.h

```diff
@@ -90,14 +95,40 @@ const verible::SyntaxTreeNode& GetUnqualifiedIdFromLocalRoot(
 const verible::SyntaxTreeNode& GetUnqualifiedIdFromReferenceCallBase(
     const verible::Symbol& reference_call_base);
 
+// Returns the node tagged with kStructType, kEnumType or kUnionType from node
+// tagged with kInstantationType.
+const verible::SyntaxTreeNode* GetStructOrUnionOrEnumTypeFromInstantiationType(
+    const verible::Symbol& instantiation_type);
+
 // Extracts kPackedDimensions node from node tagged with kDataTypePrimitive.
 const verible::SyntaxTreeNode& GetPackedDimensionFromDataType(
     const verible::Symbol& data_type_primitive);
 
 // Extracts kUnqualifiedId node from nodes tagged with kInstantiationType.
-const verible::SyntaxTreeNode& GetUnqualifiedIdFromInstantiationType(
+const verible::SyntaxTreeNode* GetUnqualifiedIdFromInstantiationType(
     const verible::Symbol& instantiation_type);
 
+// Return the type node of the given type decalration.
+//- The type can kStructType, kUnionType or kEnumType.
```

Comment:
Could generally be anything.  Anything that could be a data declaration could also be `typedef`d, like C++.